General/Presentation todo:
- deploy on PythonAnywhere (once all edits done)
- Submit slides

Website todo:
- finish test cases
- favorite recipes for each user



ERRORS



======================================================================
ERROR: test_create_pages_for_categories (recipes.tests.Chapter5ModelTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 658, in test_create_pages_for_categories
    new_category_recipe.chef=new_chef #String? object of type chef??
NameError: name 'new_chef' is not defined

======================================================================
ERROR: test_navigate_from_about_to_index (recipes.tests.NeighborhoodCookbookLiveServerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 379, in test_navigate_from_about_to_index
    self.client.get(reverse('index'))
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 536, in get
    **extra)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 340, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 416, in generic
    return self.request(**r)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 501, in request
    six.reraise(*exc_info)
  File "/anaconda3/lib/python3.7/site-packages/django/utils/six.py", line 686, in reraise
    raise value
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/views.py", line 19, in index
    dinners = Category.objects.get(name="Dinner").recipe_set.all()
  File "/anaconda3/lib/python3.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/anaconda3/lib/python3.7/site-packages/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
recipes.models.DoesNotExist: Category matching query does not exist.

======================================================================
ERROR: test_navigate_from_contactus_to_index (recipes.tests.NeighborhoodCookbookLiveServerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 409, in test_navigate_from_contactus_to_index
    self.client.get(reverse('index'))
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 536, in get
    **extra)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 340, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 416, in generic
    return self.request(**r)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 501, in request
    six.reraise(*exc_info)
  File "/anaconda3/lib/python3.7/site-packages/django/utils/six.py", line 686, in reraise
    raise value
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/views.py", line 19, in index
    dinners = Category.objects.get(name="Dinner").recipe_set.all()
  File "/anaconda3/lib/python3.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/anaconda3/lib/python3.7/site-packages/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
recipes.models.DoesNotExist: Category matching query does not exist.

======================================================================
ERROR: test_navigate_from_faqs_to_index (recipes.tests.NeighborhoodCookbookLiveServerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 439, in test_navigate_from_faqs_to_index
    self.client.get(reverse('index'))
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 536, in get
    **extra)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 340, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 416, in generic
    return self.request(**r)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 501, in request
    six.reraise(*exc_info)
  File "/anaconda3/lib/python3.7/site-packages/django/utils/six.py", line 686, in reraise
    raise value
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/views.py", line 19, in index
    dinners = Category.objects.get(name="Dinner").recipe_set.all()
  File "/anaconda3/lib/python3.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/anaconda3/lib/python3.7/site-packages/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
recipes.models.DoesNotExist: Category matching query does not exist.

======================================================================
ERROR: test_navigate_from_index_to_about (recipes.tests.NeighborhoodCookbookLiveServerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 371, in test_navigate_from_index_to_about
    about_link = self.browser.find_element_by_partial_link_text("About")
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 462, in find_element_by_partial_link_text
    return self.find_element(by=By.PARTIAL_LINK_TEXT, value=link_text)
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 978, in find_element
    'value': value})['value']
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"partial link text","selector":"About"}
  (Session info: headless chrome=72.0.3626.121)
  (Driver info: chromedriver=2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189),platform=Mac OS X 10.14.1 x86_64)


======================================================================
ERROR: test_navigate_from_index_to_contactus (recipes.tests.NeighborhoodCookbookLiveServerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 401, in test_navigate_from_index_to_contactus
    contact_us_link = self.browser.find_element_by_partial_link_text("Contact")
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 462, in find_element_by_partial_link_text
    return self.find_element(by=By.PARTIAL_LINK_TEXT, value=link_text)
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 978, in find_element
    'value': value})['value']
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"partial link text","selector":"Contact"}
  (Session info: headless chrome=72.0.3626.121)
  (Driver info: chromedriver=2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189),platform=Mac OS X 10.14.1 x86_64)


======================================================================
ERROR: test_navigate_from_index_to_faqs (recipes.tests.NeighborhoodCookbookLiveServerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 431, in test_navigate_from_index_to_faqs
    faqs_link = self.browser.find_element_by_partial_link_text("FAQs")
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 462, in find_element_by_partial_link_text
    return self.find_element(by=By.PARTIAL_LINK_TEXT, value=link_text)
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 978, in find_element
    'value': value})['value']
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"partial link text","selector":"FAQs"}
  (Session info: headless chrome=72.0.3626.121)
  (Driver info: chromedriver=2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189),platform=Mac OS X 10.14.1 x86_64)


======================================================================
ERROR: test_navigate_from_index_to_suggestcuisine (recipes.tests.NeighborhoodCookbookLiveServerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 461, in test_navigate_from_index_to_suggestcuisine
    suggest_cuisine_link = self.browser.find_element_by_partial_link_text("SuggestCuisine")
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 462, in find_element_by_partial_link_text
    return self.find_element(by=By.PARTIAL_LINK_TEXT, value=link_text)
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 978, in find_element
    'value': value})['value']
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/anaconda3/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"partial link text","selector":"SuggestCuisine"}
  (Session info: headless chrome=72.0.3626.121)
  (Driver info: chromedriver=2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189),platform=Mac OS X 10.14.1 x86_64)


======================================================================
ERROR: test_navigate_from_suggestcuisine_to_index (recipes.tests.NeighborhoodCookbookLiveServerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 469, in test_navigate_from_suggestcuisine_to_index
    self.client.get(reverse('index'))
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 536, in get
    **extra)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 340, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 416, in generic
    return self.request(**r)
  File "/anaconda3/lib/python3.7/site-packages/django/test/client.py", line 501, in request
    six.reraise(*exc_info)
  File "/anaconda3/lib/python3.7/site-packages/django/utils/six.py", line 686, in reraise
    raise value
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/anaconda3/lib/python3.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/views.py", line 19, in index
    dinners = Category.objects.get(name="Dinner").recipe_set.all()
  File "/anaconda3/lib/python3.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/anaconda3/lib/python3.7/site-packages/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
recipes.models.DoesNotExist: Category matching query does not exist.

======================================================================
ERROR: test_admin_page_contains_name_cooktime_and_chef (recipes.tests.NeighbourhoodCookbookLiveServerTests2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 551, in test_admin_page_contains_name_cooktime_and_chef
    test_utils.login(self)
NameError: name 'test_utils' is not defined

======================================================================
ERROR: test_can_create_new_category_via_admin_site (recipes.tests.NeighbourhoodCookbookLiveServerTests2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 587, in test_can_create_new_category_via_admin_site
    test_utils.login(self)
NameError: name 'test_utils' is not defined

======================================================================
ERROR: test_population_script (recipes.tests.NeighbourhoodCookbookLiveServerTests2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 516, in test_population_script
    test_utils.login(self)
NameError: name 'test_utils' is not defined

======================================================================
FAIL: test_population_script_changes (recipes.tests.Chapter5ModelTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 705, in test_population_script_changes
    self.assertEquals(cat.likes, 0)
AssertionError: 160 != 0

======================================================================
FAIL: test_about_contain_profile_image (recipes.tests.NeighbourhoodCookbookAboutPageTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 120, in test_about_contain_profile_image
    self.assertIn(b'img src="/media/profile_pics', response.content)
AssertionError: b'img src="/media/profile_pics' not found in b'<!DOCTYPE html>\n\n\n<html>\n\t<head lang="en">\n    <meta charset="utf-8" />\n\t<link rel="stylesheet" href="/static/css/main.css">\n\t<title>\n\t\t\n  About\n\n    </title>\n\t</head>\n \t<div class="header">\n\t\t<a class="base" href="/recipes/"><img src="/static/images/logo.png" alt="Website Logo" width="170"/></a>\n\t\t<a class="base" href="/recipes/trending/">Trending</a>\n\t\t\n\n\t\t\t<a class="base" href="/recipes/login/">Login</a>\n\t\t\t<a class="base" href="/recipes/register/">Sign Up</a>\n\t\t\n\t</div>\n\n\t<body>\n    <div>\n      \n<h1> About </h1>\n\n  <div>\n\n    <h2> The Neighborhood Cook Book is a website designed for people to be able to view, upload, rate and review recipes. </h2>\n    \n    <p> The website primarily focuses on allowing people to improve their cooking skills. </p>\n\n    <h4> This website was designed by Lynda, Amy, Qiufei and Eve </h4>\n  </div>\n  <p>superuser username: wad2 password: wad2teamproject</p>\n\n  <h3>Admins</h3>\n  \n    <strong>There are no admins present.</strong>\n  \n\n\n    </div>\n\t</body>\n\n  <div class="footer">\n    <a class="base" href="/recipes/about/">About the Creators</a>\n    <a class="base" href="/recipes/contact/">Contact Us</a>\n\t<a class="base" href="/recipes/faq/">FAQs</a>\n    \n  </div>\n</html>'

======================================================================
FAIL: test_index_contains_welcome_message (recipes.tests.NeighbourhoodCookbookIndexPageTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 60, in test_index_contains_welcome_message
    self.assertIn(b'Welcome', response.content)
AssertionError: b'Welcome' not found in b'<!DOCTYPE html>\n\n\n<html>\n\t<head lang="en">\n    <meta charset="utf-8" />\n\t<link rel="stylesheet" href="/static/css/main.css">\n\t<title>\n\t\t\n\tIndex\n\n    </title>\n\t</head>\n \t<div class="header">\n\t\t<a class="base" href="/recipes/"><img src="/static/images/logo.png" alt="Website Logo" width="170"/></a>\n\t\t<a class="base" href="/recipes/trending/">Trending</a>\n\t\t\n\n\t\t\t<a class="base" href="/recipes/login/">Login</a>\n\t\t\t<a class="base" href="/recipes/register/">Sign Up</a>\n\t\t\n\t</div>\n\n\t<body>\n    <div>\n      \n\t\n\t<h3>Recipes</h3>\n\t\n\t<ul>\n\t\n\t  <li>\n\t\t<a href="/recipes/q_smart-all-american-burger/">All-American Burger</a> --\n\t\t<a class="name" href="/recipes/profile/q_smart/">q_smart</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/q_smart-california-rolls/">California Rolls</a> --\n\t\t<a class="name" href="/recipes/profile/q_smart/">q_smart</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/eve_ohagan-chicken-curry/">Chicken Curry</a> --\n\t\t<a class="name" href="/recipes/profile/eve_ohagan/">eve_ohagan</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/eve_ohagan-chicken-pesto-panini/">Chicken Pesto Panini</a> --\n\t\t<a class="name" href="/recipes/profile/eve_ohagan/">eve_ohagan</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/lynda_faller-classic-hot-dogs/">Classic Hot Dogs</a> --\n\t\t<a class="name" href="/recipes/profile/lynda_faller/">lynda_faller</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/lynda_faller-easy-chicken-chow-mein/">Easy Chicken Chow Mein</a> --\n\t\t<a class="name" href="/recipes/profile/lynda_faller/">lynda_faller</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/amy_hynes-old-school-beef-tacos/">Old School Beef Tacos</a> --\n\t\t<a class="name" href="/recipes/profile/amy_hynes/">amy_hynes</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/lynda_faller-pancakes/">Pancakes</a> --\n\t\t<a class="name" href="/recipes/profile/lynda_faller/">lynda_faller</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/amy_hynes-spaaghetti-carbonara/">Spaaghetti Carbonara</a> --\n\t\t<a class="name" href="/recipes/profile/amy_hynes/">amy_hynes</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/eve_ohagan-st-paddys-shake/">St Paddy&#39;s Shake</a> --\n\t\t<a class="name" href="/recipes/profile/eve_ohagan/">eve_ohagan</a>\n\t  </li>\n\t\n\t  <li>\n\t\t<a href="/recipes/amy_hynes-traditional-tiramisu/">Traditional Tiramisu</a> --\n\t\t<a class="name" href="/recipes/profile/amy_hynes/">amy_hynes</a>\n\t  </li>\n\t\n\t</ul>\n\t\n\n\t<h3>Dinner Recipes</h3>\n\t\n\t<ul>\n\t\n\t\t<li>\n\t\t\t<a href="/recipes/q_smart-all-american-burger/">All-American Burger</a> --\n\t\t\t<a class="name" href="/recipes/profile/q_smart/">q_smart</a>\n\t\t</li>\n\t\n\t\t<li>\n\t\t\t<a href="/recipes/amy_hynes-spaaghetti-carbonara/">Spaaghetti Carbonara</a> --\n\t\t\t<a class="name" href="/recipes/profile/amy_hynes/">amy_hynes</a>\n\t\t</li>\n\t\n\t\t<li>\n\t\t\t<a href="/recipes/amy_hynes-old-school-beef-tacos/">Old School Beef Tacos</a> --\n\t\t\t<a class="name" href="/recipes/profile/amy_hynes/">amy_hynes</a>\n\t\t</li>\n\t\n\t\t<li>\n\t\t\t<a href="/recipes/lynda_faller-easy-chicken-chow-mein/">Easy Chicken Chow Mein</a> --\n\t\t\t<a class="name" href="/recipes/profile/lynda_faller/">lynda_faller</a>\n\t\t</li>\n\t\n\t\t<li>\n\t\t\t<a href="/recipes/eve_ohagan-chicken-curry/">Chicken Curry</a> --\n\t\t\t<a class="name" href="/recipes/profile/eve_ohagan/">eve_ohagan</a>\n\t\t</li>\n\t\n\t</ul>\n\t\n\n\t<h3>Categories</h3>\n\t\n\t\n\t\t\n\t\t\t<h4 class="cat">Breakfast</h4>\n\t\t\t<span>Likes: 64, ID: 1</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/breakfast.jpeg" alt="Picture of Breakfast" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Lunch</h4>\n\t\t\t<span>Likes: 32, ID: 2</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/lunch.jpeg" alt="Picture of Lunch" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Dinner</h4>\n\t\t\t<span>Likes: 16, ID: 3</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/dinner.jpeg" alt="Picture of Dinner" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Dessert</h4>\n\t\t\t<span>Likes: 16, ID: 4</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/dessert.jpeg" alt="Picture of Dessert" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Cuisines</h4>\n\t\t\t<span>Likes: 160, ID: 5</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/cuisines.jpeg" alt="Picture of Cuisines" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Special Occasions</h4>\n\t\t\t<span>Likes: 16, ID: 6</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/spec_occ.jpeg" alt="Picture of Special Occasions" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Cuisines</h4>\n\t\t\t<span>Italian, Likes: 64, ID: 7</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/italian.jpeg" alt="Picture of Italian" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Cuisines</h4>\n\t\t\t<span>American, Likes: 32, ID: 8</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/american.jpeg" alt="Picture of American" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Cuisines</h4>\n\t\t\t<span>Mexican, Likes: 16, ID: 9</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/mexican.jpeg" alt="Picture of Mexican" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Cuisines</h4>\n\t\t\t<span>Chinese, Likes: 16, ID: 10</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/chinese.jpeg" alt="Picture of Chinese" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Cuisines</h4>\n\t\t\t<span>Indian, Likes: 16, ID: 11</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/indian.jpeg" alt="Picture of Indian" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Cuisines</h4>\n\t\t\t<span>Japanese, Likes: 16, ID: 12</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/japanese.png" alt="Picture of Japanese" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Special Occasions</h4>\n\t\t\t<span>St Patrick&#39;s Day, Likes: 64, ID: 13</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/stpaddys.jpeg" alt="Picture of St Patrick&#39;s Day" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Special Occasions</h4>\n\t\t\t<span>Easter, Likes: 32, ID: 14</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/easter.jpeg" alt="Picture of Easter" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Special Occasions</h4>\n\t\t\t<span>Christmas, Likes: 16, ID: 15</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/christmas.jpeg" alt="Picture of Christmas" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Special Occasions</h4>\n\t\t\t<span>Halloween, Likes: 16, ID: 16</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/halloween.jpeg" alt="Picture of Halloween" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Special Occasions</h4>\n\t\t\t<span>4th of July, Likes: 16, ID: 17</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/4july.jpeg" alt="Picture of 4th of July" width="10%"/><br>\n\t\n\t\t\n\t\t\t<h4 class="cat">Special Occasions</h4>\n\t\t\t<span>Valentine&#39;s Day, Likes: 16, ID: 18</span><br>\n\t\t\n\t\t<img src="/media/cat_pics/valentines.jpeg" alt="Picture of Valentine&#39;s Day" width="10%"/><br>\n\t\n\t\n\n\t<h3>Reviews</h3>\n\t\n\t<ul>\n\t\n\t  <li>\n\t\t<a href="/recipes/lynda_faller-pancakes/">Pancakes</a> -- <a class="name" href="/recipes/profile/lynda_faller/">lynda_faller</a>\n\t  </li>\n\t\n\t</ul>\n\t\n\n\n    </div>\n\t</body>\n\n  <div class="footer">\n    <a class="base" href="/recipes/about/">About the Creators</a>\n    <a class="base" href="/recipes/contact/">Contact Us</a>\n\t<a class="base" href="/recipes/faq/">FAQs</a>\n    \n  </div>\n</html>'

======================================================================
FAIL: test_cuisines_cat_with_likes (recipes.tests.NeighbourhoodCookbookModelTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qsmart/School/Web App Development/wad2_team_project/recipes/tests.py", line 181, in test_cuisines_cat_with_likes
    self.assertEqual(cat.likes, 0)
AssertionError: 160 != 0

----------------------------------------------------------------------
Ran 41 tests in 42.673s

FAILED (failures=4, errors=12)
Destroying test database for alias 'default'...
